pkg/operator/ceph/object/spec.go

```diff
@@ -34,21 +34,38 @@ import (
 
 func (c *config) startDeployment() error {
 
-	deployment := &extensions.Deployment{
+	d := &extensions.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      c.instanceName(),
 			Namespace: c.store.Namespace,
 		},
 		Spec: extensions.DeploymentSpec{Template: c.makeRGWPodSpec(), Replicas: &c.store.Spec.Gateway.Instances},
 	}
-	k8sutil.SetOwnerRefs(c.context.Clientset, c.store.Namespace, &deployment.ObjectMeta, c.ownerRefs)
-	_, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.store.Namespace).Create(deployment)
-	return err
+	k8sutil.SetOwnerRefs(c.context.Clientset, c.store.Namespace, &d.ObjectMeta, c.ownerRefs)
+
+	logger.Debugf("starting mds deployment: %+v", d)
+	_, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.store.Namespace).Create(d)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create rgw deployment %s: %+v", c.instanceName(), err)
+		}
+		logger.Infof("deployment for rgw %s already exists. updating if needed", c.instanceName())
+		// There may be a *lot* of rgws, and they are stateless, so don't bother waiting until the
```

Comment:
It should be safe to let the k8s deployment update take care of this for us. As documented [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#updating-a-deployment), by default at most 25% of the instances will be down during the deployment update. We could fine-tune this with policies on the deployment in the future if needed.

Reply:
I think there is even an issue open for Rook to create PodDisruptionBudgets which kind of are the "same" direction as update strategy settings.